### PR TITLE
 feat: 🎸 reschedule a failed instruction

### PR DESCRIPTION
**Pull request by sansan**
### JIRA Link 

https://polymesh.atlassian.net/browse/DA-73 

### Changelog / Description 

- adds enpoint to reschedule a failed instruction changes/features.

### Checklist - 

- [x] New Feature ?
- [x] Updated swagger annotation (if API structure is changed) ?
- [x] Unit Test (if possible) ?
- [x] Updated the Readme.md (if required) ?

### CODE REVIEW

File: src/settlements/settlements.controller.ts [hunk 1]
```diff
@@ -133,6 +137,36 @@ export class SettlementsController {
...
+    name: 'id',
+    description: 'The ID of the Instruction to be rescheduled',
+    type: 'string',
+    example: '123',
+  })
+  @ApiOkResponse({
+    description: 'Details of the transaction',
+    type: TransactionQueueModel,
+  })
+  @ApiTransactionFailedResponse({
```
Comment (polymath-eric) [inline src/settlements/settlements.controller.ts:155]: What happens if you try to reschedule an instruction you aren't involved in?

Comment (polymath-eric) [reply src/settlements/settlements.controller.ts:155]: Is there Not Found, or is it just Unprocessable?

**Review by polymath-eric — APPROVED** [review]
LGTM.

Left a few questions around the responses, but what you have might be good.

File: src/settlements/settlements.controller.ts (hunk 1 above)
Comment (prashantasdeveloper) [reply src/settlements/settlements.controller.ts:155]: NOT_FOUND will also be there for case where instruction for given ID is not present. `findInstruction` in the service throws that error

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]

**Review by prashantasdeveloper — APPROVED** [review]

**Pull request merged**